lsp.el

```diff
@@ -807,34 +811,38 @@ WORKSPACE is the workspace that contains the diagnostics."
       (with-current-buffer buffer
         (run-hooks 'lsp-after-diagnostics-hook)))))
 
-(eval-after-load 'flymake
-  '(progn
-     (defun lsp--flymake-setup()
-       "Setup flymake."
-       (flymake-mode-on)
-       (add-hook 'flymake-diagnostic-functions 'lsp--flymake-backend nil t)
-       (add-hook 'lsp-after-diagnostics-hook 'lsp--flymake-after-diagnostics nil t))
-
-     (defun lsp--flymake-after-diagnostics ()
-       "Handler for `lsp-after-diagnostics-hook'"
-       (when lsp--flymake-report-fn
-         (lsp--flymake-backend lsp--flymake-report-fn)
-         (remove-hook 'lsp-after-diagnostics-hook 'lsp--flymake-after-diagnostics t)))
-
-     (defun lsp--flymake-backend (report-fn &rest _args)
-       "Flymake backend."
-       (funcall report-fn
-                (-some->> (lsp-diagnostics)
-                          (gethash buffer-file-name)
-                          (--map (-let (((&hash "message" "severity" "range" (&hash "start" "end")) (lsp-diagnostic-original it)))
-                                   (flymake-make-diagnostic (current-buffer)
-                                                            (lsp--position-to-point start)
-                                                            (lsp--position-to-point end)
-                                                            (case severity
-                                                              (1 :error)
-                                                              (2 :warning)
-                                                              (_ :success))
-                                                            message))))))))
+(with-eval-after-load 'flymake
+  (defun lsp--flymake-setup()
+    "Setup flymake."
+    (flymake-mode-on)
+    (add-hook 'flymake-diagnostic-functions 'lsp--flymake-backend nil t)
+    (add-hook 'lsp-after-diagnostics-hook 'lsp--flymake-after-diagnostics nil t))
+
+  (defun lsp--flymake-after-diagnostics ()
+    "Handler for `lsp-after-diagnostics-hook'"
+    (when lsp--flymake-report-fn
+      (lsp--flymake-backend lsp--flymake-report-fn)
+      (remove-hook 'lsp-after-diagnostics-hook 'lsp--flymake-after-diagnostics t)))
+
+  (defun lsp--flymake-backend (report-fn &rest _args)
+    "Flymake backend."
+    (funcall report-fn
+             (-some->> (lsp-diagnostics)
+                       (gethash buffer-file-name)
+                       (--map (-let [(&hash "message" "severity" "range") (lsp-diagnostic-original it)]
+                                (-let [(start . end) (lsp--range-to-region range)]
```

Comment:
is lsp--range-to-region really needed? Can we destruct and then assign to start and end?

Reply:
It can be used to simply other similar patterns.